cmd/opampsupervisor/Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM golang:1.24-alpine3.20
+RUN apk --update add ca-certificates
+
+ARG SERVICE_NAME=opampsupervisor
+
+RUN addgroup --gid 10001 --system ${SERVICE_NAME} && \
+    adduser --ingroup ${SERVICE_NAME} --shell /bin/false \
+    --disabled-password --uid 10001 ${SERVICE_NAME}
+
+USER ${SERVICE_NAME}
+WORKDIR /home/${SERVICE_NAME}
+
+COPY --chmod=755 opampsupervisor /usr/local/bin/opampsupervisor
```

Comment:
Should the released docker image include a collector binary? Without a binary the image is only useful as a parent image, but not being used on its own.

Reply:
I can see it both ways. Including a Collector binary in the image makes it easier to get started, but coupling the Supervisor and Collector feels odd, and I think it would also be valid to use the Supervisor image as a base image.

Users could also mount a volume with their preferred Collector binary in it, which would have the advantage of allowing Collector binaries managed through OpAMP's package management functionality to be persistent, but would have the disadvantage of not using standard application deployment mechanisms.

Reply:
the binary can always be added later on, this will at least give end users the ability to bring their own collectors with the existing image